docs/DeveloperGuide.md

```diff
@@ -1,38 +1,442 @@
+
+Sequence Diagram
+
+<img src="./DGDiagramsCreator/DGUsedDiagrams/SequenceDiagramNew.png" width="100%" />
+
+Each of the six main components (also shown in the diagram above)
+
+
+
+* defines its API in an interface with the same name as the Component.
+
+
+## UI Component
+
+The API of this component is specified in `Ui.java`
+
+<img src="./DGDiagramsCreator/DGUsedDiagrams/LightBulb.png" width="10%" />
+
+
+The UI consists of the Ui class which handles all the inputs provided by the user. The Ui then sends information to the 
+parser which will interpret the commands provided by user input. The **Parser **will create **Commands** which will be
+sent to **SecureNUS** to execute some command that will alter the data in SecretStorage. The **Ui **also handles 
+responses/ feedback to the User.
+
+
+## SecureNUS Component
+
+This is the main component that initialises all other components and connects them when the application is running.
+
+## SecretStorage Component
+
+<img src="./DGDiagramsCreator/DGUsedDiagrams/SecretStorageComponent.png" width="100%" />
+
+The Secret Storage component abstracts the data that is to be stored into many views/ indexes. This optimizes the speed of queries for many types of queries without developers needing to access the individual data structures used. The data structures, namely Hash Table and Array are stored in the SecretSearcher and SecretEnumerator class respectively. Finally, during the initialisation of the application, the Backend will connect past data to the SecretMaster. During shutdown, the Backend will handle saving it into an encrypted file.
+
+
+[//]: # (@@author stevenantya)
+## Command Component
+
+The API of this component is specified in Command.java
+
+Command is an abstract class that is inherited by various components such as AddBasicPasswordCommand.java
+
+
+
+<img src="./DGDiagramsCreator/DGUsedDiagrams/CommandComponent.png" width="100%" />
+
+The Command consist of Command abstract class that handles all of its command constructors and executions through its child classes. The user inputs command in Ui, that is parsed in Parser, and then institiated in and then executed in Command classes.
+
+
+## Backend Component
+<img src="./DGDiagramsCreator/DGUsedDiagrams/BackendComponent.png" width="100%" />
+
+
+The backend manages the saving and loading of data from the database.txt file. To do so, it collates and organises information stored in the SecretMaster and encrypts it into the database.txt. When starting up, it will decrypt the database.txt file and transfer it’s data into the SecretMaster.
+
+
+---
+
+[//]: # (@@author euzhengxi)
+# Product Scope
+
+
+## Target user profile:
+
+
+
+* has a need to manage a significant number of passwords
+* prefer desktop apps over other types
+* can type fast
+* prefers typing to mouse interactions
+* is reasonably comfortable using CLI apps
+
+
+## Value proposition:
```

Comment:
May consider defining the scope more clearly by specifying the boundary beyond which the app will not help.